doc/rtd/howto/predeploy_testing.rst

```diff
@@ -3,139 +3,69 @@
 How to test ``cloud-init`` locally before deploying
 ***************************************************
 
+How to run cloud-init locally
+=============================
 It's very likely that you will want to test ``cloud-init`` locally before
 deploying it to the cloud. Fortunately, there are several different virtual
-machines (VMs) and container tools that are ideal for this sort of local
+machine (VM) and container tools that are ideal for this sort of local
 testing.
 
-In this guide, we will show how to use three of the most popular tools:
-`Multipass`_, `LXD`_ and `QEMU`_.
+To run cloud-init locally, head over to the
+:ref:`tutorials page<tutorial_index>` page, which will show you how to run
+cloud-init locally. Once you know how to run cloud-init locally, you may wish
+to "rerun" cloud-init while developing your cloud config for faster
+development.
 
-Multipass
-=========
+.. _how_to_rerun_cloud-init:
 
-Multipass is a cross-platform tool for launching Ubuntu VMs across Linux,
-Windows, and macOS.
+How to re-run cloud-init
+========================
 
-When a user launches a Multipass VM, user data can be passed by adding the
-``--cloud-init`` flag and the appropriate YAML file containing the user data:
+Cloud-init runs on "first boot", so simply rebooting the system will
+not run cloud-init again[1]. Cloud-init provides two different options for
+rerunning cloud-init for debugging purposes.
 
-.. code-block:: shell-session
-
-    $ multipass launch bionic --name test-vm --cloud-init userdata.yaml
-
-Multipass will validate the YAML syntax of the cloud-config file before
-attempting to start the VM! A nice addition which saves time when you're
-experimenting and launching instances with various cloud-configs.
-
-Multipass *only* supports passing user data, and *only* as YAML cloud-config
-files. Passing a script, a MIME archive, or any of the other user data formats
-``cloud-init`` supports will result in an error from the YAML syntax validator.
-
-LXD
-===
-
-LXD offers a streamlined user experience for using Linux system containers.
-With LXD, a user can pass:
+.. warning::
 
-* user data,
-* vendor data,
-* metadata, and
-* network configuration.
+    Making cloud-init run again may be destructive and must never be done on a production system. Artifacts such as ssh keys or passwords may be overwritten.
 
-The following command initialises a container with user data:
+Option #1: Remove the logs and cache then reboot
```

Comment:
Think it's worth mentioning 
`cloud-init clean --logs; cloud-init init --local; cloud-init init; cloud-init modules --mode=config; cloud-init modules --mode=final; cloud-init status --wait --long`? It avoids a costly reboot so this is my default unless it's a quick-reboot like LXD.

Reply:
> cloud-init status --wait --long

Why include the --wait command? To ensure that the reporter thread cleans up?

Reply:
I guess that I could include this, however this section is intended to tell a user how to re-run ALL of cloud-init, which this will not do. I will add another section.

Reply:
> Why include the --wait command?

It was just habit from waiting for cloud-init to run when connecting into an instance. But if we run each stage explicitly here, there's no need for the last status command.